src/__swaps__/screens/Swap/components/SwapNumberPad.tsx

```diff
@@ -48,6 +48,10 @@ export const SwapNumberPad = () => {
     const inputKey = focusedInput.value;
     const currentValue = SwapInputController.inputValues.value[inputKey].toString();
     const newValue = currentValue === '0' ? `${number}` : `${currentValue}${number}`;
+    // It will overflow with 18 digits
+    if (newValue.length > 17) {
+      return;
+    }
```

Comment:
Isn't this a string?

Also, aren't there valid input amounts that this will skip?

Reply:
(Meaning where does the overflow happen?)

Reply:
Fixed and working up to 78 digits (max uint256 size)


https://github.com/rainbow-me/rainbow/assets/1247834/80ed05d1-bc5e-4b60-8c14-1300aeede46d

Reply:
side note, but uhhhhhhh should we limit this to maybe a smaller amount than 78 😅

Reply:
No

Reply:
I'm pro letting anyone dump their entire token supply